### PR TITLE
[release/2.5] enable NHWC batchnorm with MIOpen

### PR DESCRIPTION
**Pull request by dnikolaev-amd**
NHWC batchnorm on MIOpen in preview mode
supported modes:
* NCHW/NHWC fp32
* NCHW/NHWC fp16/bf16 mixed mode (with fp16 input/gradinet and fp32 scale/bias)

redundant NHWC-NCHW-NHWC conversions for MiopenBatchNormBackward is fixed

NHWC batchnorm disabled by default. Environment variable PYTORCH_MIOPEN_SUGGEST_NHWC_BATCHNORM=1 enables NHWC batchnorm (separately from NHWC convolution)

new batchnorm tests (most of them are skipped by default) (only for debugging purposes):
```
test_batchnorm_layout_NCHW_mixed_False_bfloat16 - skipped - not supported mode
test_batchnorm_layout_NCHW_mixed_False_float16 - skippes - not supported mode
test_batchnorm_layout_NCHW_mixed_False_float32 - passed
test_batchnorm_layout_NCHW_mixed_True_bfloat16 - skipped - tolerance issue
test_batchnorm_layout_NCHW_mixed_True_float16 - passed
test_batchnorm_layout_NCHW_mixed_True_float32 - skipped - useless
test_batchnorm_layout_NHWC_mixed_False_bfloat16 - skipped - NHWC
test_batchnorm_layout_NHWC_mixed_False_float16 - skippes - not supported mode
test_batchnorm_layout_NHWC_mixed_False_float32 - skippes - not supported mode
test_batchnorm_layout_NHWC_mixed_True_bfloat16 - skipped - NHWC
test_batchnorm_layout_NHWC_mixed_True_float16 - skipped - NHWC
test_batchnorm_layout_NHWC_mixed_True_float32 - skipped - useless

test_batchnorm_nhwc_eval_mixed_False_bfloat16 - skipped - not supported mode
test_batchnorm_nhwc_eval_mixed_False_float16 - skipped - not supported mode
test_batchnorm_nhwc_eval_mixed_False_float32 - skipped - NHWC
test_batchnorm_nhwc_eval_mixed_True_bfloat16 - skipped - NHWC
test_batchnorm_nhwc_eval_mixed_True_float16 - skipped - NHWC
test_batchnorm_nhwc_eval_mixed_True_float32 - skipped - useless
```

(cherry picked from commit a1e8b0e925114d34e9feb994fb76e447b6bf264d)
(cherry picked from commit 8c39322153d9608f06f69cdc588cbe0acd2d53aa)
(cherry picked from commit d726574c808284976e943858ee2ce9b40ccbb5db)
(cherry picked from commit 432b200751f5dc3eb636e0639c182029ff7af750)
(cherry picked from commit 398bd573dcde7a5f5e69c00e1e5a02fd349aee71) 
(cherry picked from commit d9668352f0d960457a40e1ea9458aa11d8668c8d)

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
Need to refactor it to bring in the changes from #2022

**Comment by jithunnair-amd** [comment]
Closing, until there's an ask to add NHWC BN support to 2.5
